### PR TITLE
Add Ruff Lint

### PR DESCRIPTION
**Pull request by JasonGrace2282**
Adds ruff linting and replaces isort in pre-commit ci

Only change I'm not sure of is in `manim/mobject/opengl/opengl_mobject.py` because there are two `invert` methods defined. I deleted the one without the `self.assemble_family`, but I'm not sure if that is the correct behavior (it's the current behavior).

### CODE REVIEW

**Review by Viicos — CHANGES_REQUESTED** [review]
Sorry had some pending comments

File: pyproject.toml [hunk 1]
```diff
@@ -134,6 +134,37 @@ extend-exclude = [
 ]
 fix = true
 
+[tool.ruff.lint]
+extend-select = [
+  "I",
+]
+
+extend-ignore = [
```
[the remark this answers is not in the document]
Comment (JasonGrace2282) [reply pyproject.toml]: Done!

File: manim/utils/docbuild/autoaliasattr_directive.py [hunk 2]
```diff
@@ -49,7 +48,9 @@ def smart_replace(base: str, alias: str, substitution: str) -> str:
     occurrences = []
     len_alias = len(alias)
     len_base = len(base)
-    condition = lambda char: (not char.isalnum()) and char != "_"
+
+    def condition(char: str):
```
Comment (Viicos) [inline manim/utils/docbuild/autoaliasattr_directive.py:52]: Let's fully type it :)

```suggestion
    def condition(char: str) -> bool:
```

**Review by Viicos — APPROVED** [review]
Great!

**Comment by Viicos** [comment]
> Only change I'm not sure of is in `manim/mobject/opengl/opengl_mobject.py` because there are two `invert` methods defined. I deleted the one without the `self.assemble_family`, but I'm not sure if that is the correct behavior (it's the current behavior).

Unfortunate, but let's keep it this way. Keeping the last method was the best path :)

**Pull request merged**